### DISCUSSION TITLE
Adding global file dependencies to specific tasks

### DISCUSSION THREAD

**Discussion by TxHawks** [vercel/turborepo#706] [Help] [answered]
Is there a way to add file-dependencies to a specific task, in the same way files can be added to `globalDependencies`?

For instance, I'd like to add the root-dir's `.eslintrc.js` file to `lint` tasks, but wouldn't want it to blow up the cache on `test` tasks.

This is pretty much the same as the documented ability to add environment vars to specific tasks, just with files outside the package dir

**Comment by nathanhammond** [comment]
Hey @TxHawks! I haven't actually tried this, but in theory you should be able to specify the `inputs` key for a particular node and pass in a traversal path to have it work.

Note that we've updated our recommendation about how to integrate with things like ESLint and would encourage you to create a package with a [shareable configuration](https://eslint.org/docs/developer-guide/shareable-configs) for your ESLint rules and then pass that package in to `extends` in your `.eslintrc.js` at the root of each of your packages. This could result in greater numbers of cache misses, but is a more-elegant solution for granularity of checks.

**Reply by nathanhammond** [reply]
(If this does not work, please let me know!)

**Reply by TxHawks** [reply]
Hi @nathanhammond 

Thanks for replying. Unfortunately specifying `inputs` outside the package directory doesn't work.

I actually opened opened a feature request issue for this at #1064 which details why a shareable configuration in a package isn't a great solution for this kind of dependencies.

**Reply by nathanhammond** [reply]
I found what may be the issue and would be treated as a bug. I'll get back to you after I poke at it for a bit.

**Reply by cinan** [reply]
i just run into this eslint config issue. I reported a [bug](https://github.com/vercel/turborepo/issues/1221) (I misunderstood how pipelines work; turbo is ok) but imo turbo+eslint deserves more examples/better default configuration in `npx create-turbo` script. I find a solution where I use global dependencies to miss cache when eslint package changes.

**Reply by WadePeterson** [reply]
Trying to make use of this feature in a large monorepo where we have packages at various levels of depth from the root of the repo.
e.g., 
```
.eslintrc
/server
  package.json
/packages
  /shared-package-1
    package.json
/client
  /apps
    /app-1
      package.json
    /app-2
      package.json
```

Is there any way to specify `inputs` relative to the repo root, instead of the package?

e.g., if each package has a `lint` command, and they all extend the `.eslintrc` that lives at the repo root.  The path to this root `.eslintrc` is different depending on how deep the packages is, so using something like `../../.eslintrc` wouldn't work.

Tried the following, which did not work:
```jsonc
{
  "$schema": "https://turborepo.org/schema.json",
  "pipeline": {
    "lint": {
      "inputs": [
        "/.eslintrc.js", // <-- how to specify this is from repo root and not relative to the package?
        "src/**"
      ]
    }
  }
}
```

**Reply by scamden** [reply]
> would encourage you to create a package with a [shareable configuration](https://eslint.org/docs/developer-guide/shareable-configs) for your ESLint rules and then pass that package in to extends in your .eslintrc.js

@nathanhammond i have exactly this setup and turbo is not busting the cache when i change a rule in the shareable config. for context the config package is a `workspace:*` dep. it seems maybe turbo is never busting the cache for any of these workspace deps actually. should i file a bug?

**Comment by TxHawks** [comment, marked as answer]
@nathanhammond Seems to be working fine in `1.2.9` using `inputs` (see #1064).
Thanks

**Reply by sppatel** [reply]
There are issues with doing this.  When running the task across all packages it somehow is leaving turbo related files behind.

```
.rw-r--r--   43 sachin.patel 17 May 13:33 f5d755f05b0789de-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f7ecce916ca7ab21/
.rw-r--r--   43 sachin.patel 17 May 17:01 f7ecce916ca7ab21-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f8d55ae17b8b97a2/
.rw-r--r--   43 sachin.patel 17 May 17:00 f8d55ae17b8b97a2-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f9c037450539cc87/
.rw-r--r--   43 sachin.patel 17 May 14:27 f9c037450539cc87-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f9dcc8ecdce8b90b/
.rw-r--r--   43 sachin.patel 17 May 17:00 f9dcc8ecdce8b90b-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f9e241797dd87054/
.rw-r--r--   43 sachin.patel 17 May 13:34 f9e241797dd87054-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f16c6fbf815ba902/
.rw-r--r--   43 sachin.patel 17 May 16:58 f16c6fbf815ba902-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f41ab217bfd85f6a/
.rw-r--r--   43 sachin.patel 17 May 16:50 f41ab217bfd85f6a-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f62cc2ecc94445b7/
.rw-r--r--   43 sachin.patel 17 May 17:00 f62cc2ecc94445b7-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f63ad401ddb1d9a0/
.rw-r--r--   43 sachin.patel 17 May 17:01 f63ad401ddb1d9a0-meta.json
drwxr-xr-x    - sachin.patel 17 May 17:02 f85f49a2ff770db1/
.rw-r--r--   44 sachin.patel 17 May 16:58 f85f49a2ff770db1-meta.json
```

I've opened https://github.com/vercel/turborepo/issues/1253
